### PR TITLE
Add container prop to AppBar

### PR DESCRIPTION
**Pull request by WiXSL**
Implements #6174

### CODE REVIEW

**Review by djhi — CHANGES_REQUESTED** [review]
As we want to keep the default behavior, I think having to pass `false` explicitly is not very intuitive. What do you think about adding a `stickOnScroll` prop which would be `false` by default? Maybe there is a better name for it though ;)

**Comment by fzaninotto** [comment]
Why don't you call this prop `container` and let users pass a Fragment instead of HideOnScroll?

File: docs/Theming.md [hunk 1]
```diff
@@ -740,9 +740,22 @@ const App = () => (
...
 
-By default, React-admin uses [Material-ui's `<AppBar>` component](https://material-ui.com/api/app-bar/) together with [react-headroom](https://github.com/KyleAMathews/react-headroom) to hide the `AppBar` on scroll. Here is an example top bar rebuilt from scratch to remove the "headroom" effect:
+```jsx
+// in src/MyAppBar.js
+import * as React from 'react';
+import { Fragment } from 'react';
+import { AppBar } from 'react-admin';
+
+const MyAppBar = props => (
+    <AppBar {...props} conatainer={Fragment} />
```
Comment (djhi) [inline docs/Theming.md:752]:
```suggestion
    <AppBar {...props} container={Fragment} />
```

Achievement unlocked: Corrected El Corrector!

**Review by djhi — CHANGES_REQUESTED** [review]
:joy: :tada:

**Review by djhi — APPROVED** [review]

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
thanks!

**Pull request merged**